### DISCUSSION TITLE
Is it possible to translate some variables like title or tagline ?

### DISCUSSION THREAD

**Discussion by clovis-lacombe** [facebook/docusaurus#4480] [Q&A] [answered]
For now, I'm pretty happy with i18n system in Docusaurus, however, I have the feeling it is not possible to translate some variables in docusaurus.config.js like title or tagline, but some other variables linked with a plugin like navbar are translatable.

My understanding is correct ?

It isn't a big deal, I can change that directly in the index.js, but I would like to confirm that I understand correctly.

Thanks for your answer

**Comment by slorber** [comment, marked as answer]
Hi,

Yes this is true that some little strings here and there are not yet translatable, and I plan to add support for the missing bits asap.

Apart that and the announcement banner, have you noticed other untranslatable strings?

**Reply by slorber** [reply]
FYI also reported here: https://github.com/facebook/docusaurus/issues/4542

**Reply by clovis-lacombe** [reply]
I have used this syntax in index.js to be able to translate the HTML body title:
`title={`${translate({id: 'homepage.title', message: 'Website title'})}`}`

I don't detect extra untranslatable strings

**Reply by alexzhc** [reply]
hi, @clovis-lacombe.  So far,  homepage title and tagline are still untranslatable by i18n.  How did you translate them in index.json using `title={${translate({id: 'homepage.title', message: 'Website title'})}}` ?  Many thanks.

**Reply by caphesuagey** [reply]
Also interested in this.

**Reply by clovis-lacombe** [reply]
Hi @alexzhc and @RodinGolodin  ,

This is the source code we have written for that: https://github.com/Skeptical-Tantra/skeptical-tantra/blob/main/src/pages/index.js#L64
But, now I'm checking better the Website, in fact, this piece of code works only partially for the homepage, not others pages.

If @slorber has any clue to be able to translate data in index.json, we are interested in.

Kind regards

**Reply by slorber** [reply]
We don't have proper support for it yet, and you'd have to swizzle many components to add similar logic to all the pages if you wanted to do it everywhere => not convenient

We'll mention when it will be supported in https://github.com/facebook/docusaurus/issues/4542

**Reply by clovis-lacombe** [reply]
Ok, thanks waiting for #4542 implementation then :-)

**Reply by mukhammed2021** [reply]
can I translate the title, tagline now?

**Reply by slorber** [reply]
The workaround shared here works: https://github.com/facebook/docusaurus/issues/4542#issuecomment-1284016559

```js
function getSiteTagline() {
  switch(process.env.DOCUSAURUS_CURRENT_LOCALE) {
    case "fr": return "Mon site en Français";
    default: return "My English website";
  }
}
```
